WordPress/gutenberg
# Input Field Block: Use `useblockProps` hook in save function

Fixes #56458
Follow up #44214

## What?

This PR adds the missing wrapper div and `blockProps` in the save function of the Input Filed (Form Input) block. This allows block support styles such as margins, borders, etc. to work on the front end.

## Why?
Without this hook, block support styles will not be applied. Also, the markup in the editor and frontend doesn't match.

## How?
At the same time as adding a hook to the save function, I also added a block deprecation.

## Testing Instructions

- Switch to trunk branch.
- Paste the markup below into your code editor and save your post.
  ```html
  
  <form class="wp-block-form" enctype="text/plain">
  
  	
  	<label class="wp-block-form-input__label">
  		<span class="wp-block-form-input__label-content">Label</span>
  		<input class="wp-block-form-input__input" type="text" name="label" aria-required="false"/></label>
  	
  
  	
  	<div class="wp-block-form-submit-button">
  		
  		<div class="wp-block-buttons">
  			
  			<div class="wp-block-button">
  				<button type="submit" class="wp-block-button__link wp-element-button">Submit</button>
  			</div>
  			
  		</div>
  		
  	</div>
  	
  </form>
  
  ```
- Switch to this PR branch.
- Reload the browser.
- You should see "**Updated Block: core/form-input**" displayed in your browser console.
- Add margin and border to this block and save the post.
- The block should not break even if you reload your browser.
- In the front end, the block support style should be applied correctly.
